Adding benchmark chart to readme (#1442)
Azure/azure-sdk-for-c: sdk/docs/iot/README.md

 Retry Policies | &radic;* | The Azure SDK for Embedded C provides guidelines for retries, but actual retries should be handled by the application.
 [IoT Plug and Play](https://docs.microsoft.com/azure/iot-pnp/overview-iot-plug-and-play) | &radic; | IoT Plug and Play enables solution builders to integrate smart devices with their solutions without any manual configuration.

> <br>
> 

Reviewer: This shouldn't be necessary. GitHub renders new lines fine. If you were using VSCode and its Markdown Preview, that doesn't handle tables well, and makes it seem like an extra new line is needed, causing the author to think they need to add more new lines like this. That ends up looking odd, due to the extra spacing:
![image](https://user-images.githubusercontent.com/6527137/96674434-a5b4c900-131d-11eb-8fc0-2def648899f5.png)

```suggestion
```

## Size Chart

> The following chart shows the RAM and ROM usage for the PIC24-IOT dev kit from Microchip found in https://github.com/Azure-Samples/Microchip-PIC-IoT-Wx compiled using Microchip’s XC16 version 1.60 compiler with the following macros: AZ_NO_LOGGING, AZ_NO_CONDITION_CHECKING and NDEBUG.

Reviewer: What is `NDEBUG`?

Also, did you mean `AZ_NO_PRECONDITION_CHECKING`?

The CMake options for those are as follows:
https://github.com/Azure/azure-sdk-for-c#cmake-options

```suggestion
The following chart shows the RAM and ROM usage for the PIC24-IOT dev kit from Microchip found in https://github.com/Azure-Samples/Microchip-PIC-IoT-Wx compiled using Microchip’s XC16 version 1.60 compiler with the following  compilation, preprocessor options: AZ_NO_LOGGING and AZ_NO_PRECONDITION_CHECKING.
```

This PIC24-IOT kit has 2 MCUs (microcontrollers):

>  - Host MCU, a 16-bit **PIC24FJ128GA705** with 128 KB Flash Memory & 16 KB SRAM: this MCU runs the Embedded C SDK, the sample code and the MQTT client provided by Microchip.

Reviewer: Add  a new line between text and list, to be consistent with below.
```suggestion

 - Host MCU, a 16-bit **PIC24FJ128GA705** with 128 KB Flash Memory & 16 KB SRAM: this MCU runs the Embedded C SDK, the sample code and the MQTT client provided by Microchip.
```

>  - Wi-fi MCU, an **ATWINC**1510 Wi-Fi Module: this MCU runs the TLS and TCP stack. 

Reviewer: ```suggestion
 - Wi-fi MCU, an **ATWINC**1510 Wi-Fi Module: this MCU runs the TLS and TCP stack.
```

> The Pic24 sample includes both Hub and DPS services. The table below shows RAM/ ROM sizes considering:

Reviewer: ```suggestion
The Pic24 sample includes both Hub and DPS services. The table below shows RAM/ROM sizes considering:
```

> The Pic24 sample includes both Hub and DPS services. The table below shows RAM/ ROM sizes considering:

Reviewer 2: I would reword to:

```suggestion
The chart considers the following two size comparisons:  

- Embedded C SDK size -  This refers only to the libraries of the Embedded C SDK.  It is the baseline size.
- Total size - This includes both the Embedded C SDK libraries and also the Microchip MQTT client, which has a separate Wi-Fi module running the TLS and TCP/IP stacks.
```

> -  Embedded C SDK libraries only – which represent the baseline size 

Reviewer: ```suggestion
-  Embedded C SDK libraries only – which represent the baseline size.
```

-  Total size – which includes the Embedded C SDK and Microchip MQTT client (this Dev Kit has a separate Wi-Fi module which runs the TLS and TCP/IP stacks).

|  | Embedded C SDK size | | Total Size | |
|---------|----------|---------|---------|---------
|**Sample** | **Program/ROM** | **Data/RAM** | **Program/ROM** | **Data/RAM** | 
| PIC24 (Hub + DPS + IoT Plug and Play) | 26.15KB | 0 | 103.61KB | 10.57KB

> | PIC24 Telemetry only | 2.58KB | 0 | 74.16KB | 8.26KB

Reviewer: Just seeing absolute numbers, the question comes to mind: Are these numbers good? What's the goal or comparison to the alternative/previous, which would highlight the value of the new Embedded SDK?

Author: Comparison shows the memory usage when you have only telemetry x all features. It highlights how small of a footprint can be used in a device with the Embedded C SDK.

> <br>This table will be expanded as new Hardware samples become available.<br>

Reviewer: Is the `<br>` necessary? The extra new line seems odd when rendered on GitHub.

```suggestion
This table will be expanded as new hardware samples become available.
```

> 

Reviewer: ```suggestion
```
